src/app/common/BUILD.gn

```diff
@@ -14,21 +14,41 @@
 
 import("//build_overrides/chip.gni")
 
+config("includes") {
+  include_dirs = [ "${chip_root}/zzz_generated/app-common" ]
+}
+
 static_library("cluster-objects") {
   output_name = "libClusterObjects"
 
   sources = [
-    "${chip_root}/zzz_generated/app-common/app-common/zap-generated/cluster-enums-check.h",
-    "${chip_root}/zzz_generated/app-common/app-common/zap-generated/cluster-enums.h",
     "${chip_root}/zzz_generated/app-common/app-common/zap-generated/cluster-objects.cpp",
     "${chip_root}/zzz_generated/app-common/app-common/zap-generated/cluster-objects.h",
   ]
 
   public_deps = [
+    "${chip_root}/src/app/data-model",
```

Comment:
This needs to have `:enums`, because the headers here include headers from there, and my understanding is that that is exactly what public_deps is for.

Reply:
I did not find that:

```
❯ rg cluster-enums zzz_generated/ 
zzz_generated/app-common/app-common/zap-generated/cluster-enums-check.h
22:#include <app-common/zap-generated/cluster-enums.h>
```

Reply:
Actual usages:

```
🐟 ❯ rg cluster-enums-check --glob '*.cpp' --glob '*.h'
src/app/clusters/time-format-localization-server/time-format-localization-server.cpp
24:#include <app-common/zap-generated/cluster-enums-check.h>
                                                                                                                                                                                  
src/app/data-model/Decode.h
21:#include <app-common/zap-generated/cluster-enums-check.h>
```

So we need data-model _to have them and whatever time format localization includes (and this is why the chip_data_model needs these ...)

Reply:
Actually I forgot about non-check ones. Lots of things in examples, but outside that:

```
🐟 ❯ rg cluster-enums --glob '*.cpp' --glob '*.h' --files-with-matches src zzz_generated/
src/platform/Tizen/NetworkCommissioningEthernetDriver.cpp
src/include/platform/NetworkCommissioning.h
src/lib/core/ClusterEnums.h
src/app/icd/server/ICDManager.h
src/app/data-model/Decode.h
src/app/CompatEnumNames.h
src/app/server/Dnssd.cpp
zzz_generated/app-common/app-common/zap-generated/cluster-enums-check.h
src/app/clusters/dishwasher-alarm-server/dishwasher-alarm-server.cpp
src/app/clusters/identify-server/identify-server.h
src/app/clusters/time-format-localization-server/time-format-localization-server.cpp
src/app/clusters/refrigerator-alarm-server/refrigerator-alarm-server.cpp
src/app/clusters/time-synchronization-server/time-synchronization-server.cpp
src/app/clusters/valve-configuration-and-control-server/valve-configuration-and-control-delegate.h
src/app/clusters/laundry-dryer-controls-server/laundry-dryer-controls-server.cpp
src/app/clusters/messages-server/messages-server.h
src/app/clusters/boolean-state-configuration-server/boolean-state-configuration-delegate.h
src/app/clusters/pump-configuration-and-control-server/pump-configuration-and-control-server.cpp
src/app/clusters/level-control/level-control.h
src/app/util/af-types.h
```

Reply:
Ah, I see the issue.  So the problem is that we have these includes:

```
zzz_generated/app-common/app-common/zap-generated/cluster-objects.h:#include <lib/core/ClusterEnums.h>

src/lib/core/ClusterEnums.h:#include <app-common/zap-generated/cluster-enums.h>

```

So indeed, no need for :enums here, the dep on lib/core is already there, and lib/core has the public dep on :enums, so we are good.

Reply:
That include is actually questionable ... we basically rely on a sub-include. Should be fixed, but probably not in this PR.